tools/ci/check_grep2.py

```diff
@@ -0,0 +1,519 @@
+from pathlib import Path
+import re
+import sys
+import time
+from collections import OrderedDict
+
+RED = "\033[0;31m"
+GREEN = "\033[0;32m"
+BLUE = "\033[0;34m"
+NC = "\033[0m"  # No Color
+
+MAP_COMMENT = "//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE"
+IGNORE_515_PROC_MARKER = "__byond_version_compat.dm"
+
+#
+# Basic map issues
+#
+# Basic map issues operate on a line-by-line basis.
+#
+def check_tgm(filename, lines):
+    for line in lines:
+        if re.match(r"^\".+\" = \(.+\)", line):
+            return "ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!"
+
+
+def check_comments(filename, lines):
+    for line in lines:
+        if (
+            re.match(r"//", line)
+            and not re.match(MAP_COMMENT, line)
+            and not re.match("name|desc", line)
+        ):
+            return "ERROR: Unexpected commented out line detected in this map file. Please remove it."
+
+
+def check_conflict_markers(filename, lines):
+    for line in lines:
+        if "Map Merge Marker" in line:
+            return "ERROR: Merge conflict markers detected in map, please resolve all merge failures!"
+
+
+def check_conflict_marker_object(filename, lines):
```

Comment:
same for this function and all others below it

Reply:
As above.